scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -441,150 +428,106 @@ public long getRunningJobsCount() {
+            Query query = session.getNamedQuery("getMeanJobPendingTime");
+            Double result = (Double) query.uniqueResult();
+            return result == null ? 0 : result;
         });
     }
 
     public double getMeanJobExecutionTime() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobExecutionTime");
-                Double result = (Double) query.uniqueResult();
-                return result == null ? 0 : result;
-            }
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobExecutionTime");
+            Double result = (Double) query.uniqueResult();
+            return result == null ? 0 : result;
         });
     }
 
     public double getMeanJobSubmittingPeriod() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobSubmittingPeriod");
-                Object[] result = (Object[]) query.uniqueResult();
-                Long count = (Long) result[0];
-                Long minSubmittedTime = (Long) result[1];
-                Long maxSubmittedTime = (Long) result[2];
-                if (count < 2) {
-                    return 0d;
-                } else {
-                    return (maxSubmittedTime - minSubmittedTime) / (double) (count - 1);
-                }
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobSubmittingPeriod");
+            Object[] result = (Object[]) query.uniqueResult();
+            Long count = (Long) result[0];
+            Long minSubmittedTime = (Long) result[1];
+            Long maxSubmittedTime = (Long) result[2];
+            if (count < 2) {
+                return 0d;
+            } else {
+                return (maxSubmittedTime - minSubmittedTime) / (double) (count - 1);
             }
-
         });
     }
 
     public long getJobRunningTime(final String jobId) {
         final long id = Long.parseLong(jobId);
 
-        Long result = executeReadOnlyTransaction(new SessionWork<Long>() {
-            @Override
-            public Long doInTransaction(Session session) {
-                JobData jobData = session.get(JobData.class, id);
-                if (jobData == null) {
-                    return null;
-                }
-                if (jobData.getFinishedTime() > 0) {
-                    return jobData.getFinishedTime() - jobData.getStartTime();
-                } else {
-                    return null;
-                }
+        Long result = executeReadOnlyTransaction(session -> {
+            JobData jobData = session.get(JobData.class, id);
+            if (jobData == null) {
+                return null;
```

Comment:
Ideally you return an EMPTY JobData object, then the caller doesn't need to handle null